pkg/codegen/go/gen.go

```diff
@@ -1642,12 +1647,14 @@ func (pkg *pkgContext) genResource(w io.Writer, r *schema.Resource, generateReso
 	}
 
 	// Setup secrets
-	if len(secretProps) > 0 {
-		for _, p := range secretProps {
+	if len(secretInputProps) > 0 {
+		for _, p := range secretInputProps {
 			fmt.Fprintf(w, "\tif args.%s != nil {\n", Title(p.Name))
 			fmt.Fprintf(w, "\t\targs.%[1]s = pulumi.ToSecret(args.%[1]s).(%[2]s)\n", Title(p.Name), pkg.outputType(p.Type))
 			fmt.Fprintf(w, "\t}\n")
 		}
+	}
```

Comment:
Nit: I don't think the `len(secretInputProps) > 0` check is necessary in this case.

```suggestion
	for _, p := range secretInputProps {
		fmt.Fprintf(w, "\tif args.%s != nil {\n", Title(p.Name))
		fmt.Fprintf(w, "\t\targs.%[1]s = pulumi.ToSecret(args.%[1]s).(%[2]s)\n", Title(p.Name), pkg.outputType(p.Type))
		fmt.Fprintf(w, "\t}\n")
	}
```